### PR TITLE
Enable client awareness in trace

### PR DESCRIPTION
**Pull request by juusopiikkila**
This adds checking for client headers in request

### CODE REVIEW

**Comment by juusopiikkila** [comment]
ok test added

File: src/index.ts [hunk 1]
```diff
@@ -70,6 +70,19 @@ export default fp(
       context.__traceBuilder = traceBuilder
 
       const body: any = context.reply?.request?.body
+      const headers = context.reply?.request?.headers
+
+      if ('apollographql-client-name' in headers) {
+        traceBuilder.trace.clientName =
+          typeof headers['apollographql-client-name'] === 'string'
```
Comment (simoneb) [inline src/index.ts:77]: I don't think headers can be anything other than strings, so this (and the one below) condition can be simplified

Comment (juusopiikkila) [reply src/index.ts:77]: Actually the type says that a header value could be string array also

Comment (simoneb) [reply src/index.ts:77]: ah ok, I didn't know that!

**Review by simoneb — APPROVED** [review]

**Pull request merged**